Source/WebCore/dom/Document.cpp

```diff
@@ -7696,6 +7696,26 @@ Document* Document::mainFrameDocument() const
     return document;
 }
 
+RefPtr<Document> Document::sameOriginTopLevelTraversable() const
```

Comment:
I'd call this `mainFrameDocumentIfSameOrigin` or something.
The current function name makes me think that it would return the highest ancestor frames which is of the same origin, not return nullptr when the main frame's origin differs.

Reply:
Correct - This name is purposefully derived from spec language, and as such when to use it should (hopefully) be very obv.